brian2/parsing/sympytools.py

```diff
@@ -148,7 +141,11 @@ def sympy_to_str(sympy_expr):
     str_expr : str
         A string representing the sympy expression.
     '''
-    
+    if sympy_expr in _sympy_to_str_cache:
+        return _sympy_to_str_cache[sympy_expr]
```

Comment:
Not sure what you mean -- this is still all about caching in memory, so I don't see how version changes can come into play.